airflow/models/mappedoperator.py

```diff
@@ -646,6 +643,8 @@ def unmap(self, resolve: None | Mapping[str, Any] | tuple[Context, Session]) ->
             # is prepended, and '__N' appended to deduplicate). This is hacky,
             # but better than duplicating the whole mangling logic.
             op.task_id = self.task_id
+            op.is_setup = kwargs.pop("is_setup", False)
+            op.is_teardown = kwargs.pop("is_teardown", False)
```

Comment:
`kwargs` is already passed to the operator’s constructor above, the pop calls should happen before that.

Reply:
yes, thank you, this was a "cleanup" that shouldn't have been cleaned up